R/kompletthet.R

```diff
@@ -272,33 +264,28 @@ beregn_kompletthet_datasett = function(data) {
 #'   ukjent_datasett = ukjent_datasett_eksempel
 #' )
 erstatt_ukjent_for_datasett = function(data, ukjent_datasett) {
-  data_uten_ukjent = tibble::tibble_row()
-
   hjelpefunksjon_na_vektor =
-    purrr::possibly(~ (.x %>% filter(variabel == .y) %>%
-      select(
-        where(~ !all(is.na(.x))),
-        -variabel
-      ) %>%
-      pull()), otherwise = NULL)
-
-  for (i in 1:ncol(data)) {
-    variabel_i = names(data)[i]
-    na_vektor_i = hjelpefunksjon_na_vektor(
-      ukjent_datasett,
-      variabel_i
+    purrr::possibly(
+      .f = \(d, var) (d |>
+        filter(variabel == var) |>
+        select(
+          where(\(x) !all(is.na(x))),
+          -variabel
+        ) |>
+        pull()),
+      otherwise = NULL
     )
 
-    d = erstatt_ukjent(
-      data = data %>% select(!!variabel_i),
-      variabel = variabel_i,
-      na_vektor = na_vektor_i
-    )
-
-    data_uten_ukjent = data_uten_ukjent %>%
-      bind_cols(d)
-  }
-  data_uten_ukjent
+  data |>
+    names() |>
+    purrr::map(
+      \(var) erstatt_ukjent(
+        data = data[var],
```

Comment:
Dette vil ikkje fungera rett for ein av data.frame og tibble. Funksjonen `[` oppfører seg forskjellig (for data.frame tar han ut verdiane i kolonna, og for tibble tar han ut ein éinkolonne-tibble). Må fiksast + leggast til testar på at ting funkar både for data.frame og tibble (testane først).

Reply:
Flyttar dette til eiga grein/PR, og fiksar der.